src/Datasource/QueryTrait.php

```diff
@@ -87,8 +87,8 @@
      * When called with a Table argument, the default table object will be set
      * and this query object will be returned for chaining.
      *
-     * @param \Cake\Datasource\RepositoryInterface|null $table The default table object to use
-     * @return \Cake\Datasource\RepositoryInterface|$this
+     * @param \Cake\Datasource\RepositoryInterface|\Cake\ORM\Table|null $table The default table object to use
```

Comment:
The param is type hinted as RepositoryInterface and Table implements that interface, so why is explicitly specifying Table necessary?

Reply:
Because it is used in other places with the concrete one (code smell that cant be fixed until the next major).